### DISCUSSION TITLE
2024-07-26 금요일

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#105] [기록보드]
2024-07-26 금요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | <img src="https://github.com/user-attachments/assets/d55a264f-95d1-4996-a561-240641c19156" width="300" height="300"/> |
| 10:00 ~ 11:00 | 기술면접 준비 | |
| 11:00 ~ 13:00 | 집중 개발 time | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 본인인증 외부 API 응답값 확인 및 테스트

// 사진
<img src="이미지 주소" width="300" height="300"/>

### 트러블 슈팅

- user id를 uuid로 받다 보니 segment가 지저분 해짐 -> login_id를 추가로 받는 것으로 해결

### 내일 계획

- supabase auth 비밀번호 찾기 기능 연결
```
```
// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>
```

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 지역 선택 페이지 재사용 가능한 컴포넌트로 분리 진행 중
- 글 작성 페이지 기능 구현
- 프로젝트 기획 문서 작성 (7/29월 참여기업 튜터 피드백 준비)
  - 취업시장에서의 유의미한 기술 사용 여부
  - 주제와 기획이 실제 비즈니스 문제를 해결 할 수 있는지 여부
  - 프로젝트 목표와 기대효과의 명확한 정의
  - 사용자 경험을 고려한 기획 여부

<table>
  <tr>
    <th>지역 선택 페이지 분리 중</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/81a62b71-e79c-4c10-8684-94d2bafee769" width="300" height="300"/></td>
  </tr>
</table>

### 트러블 슈팅

- 객체와 배열이 계속해서 섞여 있는 locationData 자료구조 이해에 큰 어려움을 겪음. (자료구조 자체보다는 타입 지정에)
- 재사용 가능한 컴포넌트로 만들면 해당 컴포넌트에서 생성해 낸 반환값을 state에 담고, 그 컴포넌트롤 호출한 페이지에서 state의 값을 사용할 수 있는가? 
  - -> 부모-자식 관계가 아니기에 props 전달이 불가능함.
  - -> zustand 등으로 전역 상태로 만들면 됨.
  - -> 그러나 전역 상태로 관리하면, 이 컴포넌트를 재사용 하는 곳, 즉 검색 페이지와 글쓰기 페이지에서 동시에 사용하고 있는데 state가 공유되면서 검색 페이지에서 생성한 state가 글쓰기 페이지에서도 그 값이 사용되면서 default 값으로 사용될 것 같음.
  - -> 전역상태 관리 하지 않고 어떻게 페이지에서 하위 컴포넌트에서의 state 값을 받아올 수 있는가?
  - -> Funnel Pattern을 커스텀 훅으로 만든 것처럼 커스텀 훅으로 만들어서 state와 컴포넌트를 둘 다 리턴해서 사용하면 됨.
  - -> 글쓰기 페이지 - 검색 페이지 간 중복되는 인원 설정 버튼과 날짜 선택 버튼, 지역 선택 리스트 등은 reusable component가 아니라 custom hook으로 제작해야 할 것 같아서 그렇게 리팩토링 중

### 내일 계획

- 글쓰기 페이지 UI 전부 작성
- 마이페이지 UI 작성 및 기능 연결

**Comment by eunohhh** [comment]
## 오은 일일 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | |
| 10:00 ~ 11:00 | 기술면접 준비 | |
| 11:00 ~ 13:00 | 집중 개발 time | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

### 오늘 한 일

- trips 페이지 UI 작성
- usePreferTheme 훅 작성
- 인증/인가 tanstack query 로 재작성 중

### 트러블 슈팅

- 인증/인가 함수들을 전부 tanstack 으로 빼야만 하는가...

### 내일 계획

- 주말 불태우자...

**Comment by slsl2** [comment]
## 개인 기록보드 서식 (김민지)

### 오늘 한 일

- 채팅 페이지 UI 작업
- supabase realtime, next.js 사용해서 채팅 구현하는 강의 서치 & 시청

### 문제 겪었던 부분

- supabase.ts의 타입 설정을 import 해오는 작업에서 살짝 헤맴
- 다른 팀원 분의 코드 보고 응용해서 작업 완료.. 감사합니다

### 주말 계획

- 채팅 기능 어느 정도 구현하기

**Comment by hyunjeongkwak** [comment]
**오늘 한 일**
- 컴포넌트, 페이지 이동 정리
- 검색 결과 페이지 UI 제작 중

**겪은 문제**
- 달력 라이브러리 사용에서 선택 날짜 데이터 가져와지지 않는 문제
- 페이지들끼리의 대치가 안되는 문제 등 -> 중간 발표가 얼마 남지 않아, 우선 원페이지로 구현하기로 함

**주말 할 일**
- 검색, 검색 결과 페이지 UI, 페이지 이동 정리(원페이지 : 아코디언으로 구현!)
- 칩 바꿔 끼우기
- 레이아웃 적용 시키기
